controllers/metal3.io/host_state_machine.go

```diff
@@ -183,6 +183,7 @@ func (hsm *hostStateMachine) ensureRegistered(info *reconcileInfo) (result actio
 	case metal3v1alpha1.StateRegistering:
 	default:
 		needsReregister = (hsm.Host.Status.ErrorType == metal3v1alpha1.RegistrationError ||
+			hsm.Host.Status.ErrorType == metal3v1alpha1.AdoptionError ||
```

Comment:
I've convinced myself this is not required. If it fails in adoption then registration would have succeeded just prior, so we don't need to report that we're starting a new registration.

Reply:
Yea I wasn't sure but after testing you're right, we don't need this.